### PR TITLE
Make gRPC CaptureClient independent from OrbitGl project

### PR DESCRIPTION
**Pull request by kisenshi**
### Move CaptureClient classes to new OrbitCaptureClient project.

Refactoring code related to CaptureClient so it is independent from OrbitGl. This is necessary for the creation of a new client able to handle capture requests without depending on the OrbitGl project.

CaptureClient depends on CaptureListener and CaptureEventProcessor, so it was needed to move these classes as well. Because these classes exist and are all related to a capture and not just gRPC calls, the project has been given the name of OrbitCaptureClient.

### CODE REVIEW

File: OrbitCaptureClient/CaptureListener.h [hunk 1]
```diff
@@ -2,8 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#ifndef ORBIT_GL_CAPTURE_LISTENER_H_
-#define ORBIT_GL_CAPTURE_LISTENER_H_
```
Comment (dimitry-) [inline OrbitCaptureClient/CaptureListener.h:6]: For new components please follow the structure as it is in OrbitBase - please include files under indlude/<component_name>/File.h

File: OrbitCaptureClient/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  PUBLIC CaptureClient.h
+         CaptureEventProcessor.h
+         CaptureListener.h)
+
+target_sources(
+  OrbitCaptureClient
+  PRIVATE CaptureClient.cpp
+          CaptureEventProcessor.cpp)
+
+target_include_directories(OrbitCaptureClient PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
```
Comment (dimitry-) [inline OrbitCaptureClient/CMakeLists.txt:23]: and change this to ${CMAKE_CURRENT_SOURCE_DIR}/include

File: OrbitCaptureClient/CaptureListener.h (hunk 1 above)
Comment (dimitry-) [reply OrbitCaptureClient/CaptureListener.h:6]: The idea is to use inculde/ directory for public includes and source directory for private ones. This way the file which is not supposed to be used from the outside will not be on the include path.

Comment (kisenshi) [reply OrbitCaptureClient/CaptureListener.h]: I have fixed this, thanks for letting me know and the explanation

File: OrbitCaptureClient/CMakeLists.txt (hunk 2 above)
Comment (kisenshi) [reply OrbitCaptureClient/CMakeLists.txt]: Done. I have used {CMAKE_CURRENT_LIST_DIR} instead as it is done in OrbitBase and OrbitLinuxTracing

File: OrbitCaptureClient/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2020 The Orbit Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+cmake_minimum_required(VERSION 3.15)
```
[the remark this answers is not in the document]
Comment (kisenshi) [reply OrbitCaptureClient/CMakeLists.txt]: Makes sense; I removed it. Thanks!

**Comment by kisenshi** [comment]
Fixed structure as requested.

**Pull request merged**